src/foam/core/XML.js

```diff
@@ -0,0 +1,466 @@
+/**
+@license
+Copyright 2017 The FOAM Authors. All Rights Reserved.
+http://www.apache.org/licenses/LICENSE-2.0
+*/
+
+foam.CLASS({
+  refines: 'foam.core.Property',
+
+  properties: [
+    { class: 'String', name: 'shortName' },
+    {
+      name: 'fromXML',
+      value: function fromXML(value, ctx, prop, xml) {
+        return foam.xml.parse(value, null, ctx);
+      }
+    },
+    {
+      name: 'toXML',
+      value: function toXML(value, Outputter) { return value; }
+    }
+  ],
+
+  methods: [
+    function outputXML(o) {
+      o.output({ class: '__Property__', forClass_: this.forClass_ });
+    }
+  ]
+});
+
+/** Add toXML() method to FObject. **/
+foam.CLASS({
+  refines: 'foam.core.FObject',
+
+  methods: [
+    /**
+      Output as a pretty-printed XML-ish String.
+      Use for debugging/testing purposes. If you want actual
+      XML output, use foam.xml.* instead.
+    */
+    function stringify() {
+      return foam.xml.Pretty.stringify(this);
+    }
+  ]
+});
+
+
+/** XML Outputter **/
+foam.CLASS({
+  package: 'foam.xml',
+  name: "Outputter",
```

Comment:
Can we make an abstract base class that both JSON and XML share?  There is a lot of code overlap between the two.